main.go

```diff
@@ -281,29 +252,86 @@ func main() {
 		},
 	}
 
+	// Let's Encrypt TLS mode
+	if letsencrypt {
+		// autocert
+		certmanager := autocert.Manager{
+			Prompt: autocert.AcceptTOS,
+			Cache:  autocert.DirCache(filepath.Join(datadir, "letsencrypt")),
+			HostPolicy: func(_ context.Context, host string) error {
+				host = strings.TrimPrefix(host, "www.")
+
+				if host == httpHost {
+					return nil
+				}
+
+				if host == config.FindInfo().Domain {
+					return nil
+				}
+				return fmt.Errorf("autocert: host %q not permitted by HostPolicy", host)
+			},
+		}
+
+		tlsConfig.GetCertificate = certmanager.GetCertificate
+
+		// http redirect to https and Let's Encrypt auth
+		go func() {
+			redir := httprouter.New()
+			redir.GET("/*path", func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+				r.URL.Scheme = "https"
+				r.URL.Host = httpHost
+				http.Redirect(w, r, r.URL.String(), http.StatusFound)
+			})
+
+			httpd := &http.Server{
+				Handler:        certmanager.HTTPHandler(redir),
+				Addr:           net.JoinHostPort(httpIP, "80"),
+				WriteTimeout:   httpTimeout,
+				ReadTimeout:    httpTimeout,
+				MaxHeaderBytes: maxHeaderBytes,
+			}
+
+			if err := httpd.ListenAndServe(); err != nil {
+				logger.Fatalf("http server on port 80 failed: %s", err)
+			}
+		}()
+	}
+
+	// Using a certificate/key pair from disk
+	if tlsCertificate != "" && tlsKey != "" {
+		var certificates []tls.Certificate
+		cert, err := tls.LoadX509KeyPair(tlsCertificate, tlsKey)
+
+		if err != nil {
+			logger.Fatal(err.Error())
+			os.Exit(20)
```

Comment:
This is redundant as `logger.Fatal` makes a call to `os.Exit` by default with an exit code of 1

https://golang.org/src/log/log.go?s=10156:10184#L320